cfxdb/xbrmm/schema.py

```diff
@@ -0,0 +1,127 @@
+##############################################################################
+#
+#                        Crossbar.io FX
+#     Copyright (C) Crossbar.io Technologies GmbH. All rights reserved.
+#
+##############################################################################
+
+from cfxdb.xbrmm.consent import Consents, IndexConsentByMemberAddress
+
+from cfxdb.xbrmm.channel import PaymentChannels, IndexPaymentChannelByDelegate, \
+    PaymentChannelBalances, PayingChannels, IndexPayingChannelByDelegate, \
+    PayingChannelBalances
+
+from cfxdb.xbrmm.offer import Offers, IndexOfferByKey
+from cfxdb.xbrmm.transaction import Transactions
+
+
+class Schema(object):
+    """
+    CFC edge database schema for ZLMDB.
+    """
+    def __init__(self, db):
+        self.db = db
+
+    consents: Consents
+    """
+    XBR data consents.
+    """
+
+    idx_consent_by_member_adr: IndexConsentByMemberAddress
+    """
+    Consents-by-members-address index with ``(member_adr|bytes[20], joined|int) -> member_adr|UUID`` mapping.
+    """
+
+    payment_channels: PaymentChannels
+    """
+    Payment channels for XBR consumer delegates.
+    """
+
+    idx_payment_channel_by_delegate: IndexPaymentChannelByDelegate
+    """
+    Maps from XBR consumer delegate address to the currently active payment
+    channel address for the given consumer delegate.
+    """
+
+    payment_balances: PaymentChannelBalances
+    """
+    Current off-chain balances within payment channels.
+    """
+
+    paying_channels: PayingChannels
+    """
+    Paying channels for XBR provider delegates.
+    """
+
+    idx_paying_channel_by_delegate: IndexPayingChannelByDelegate
+    """
+    Maps from XBR provider delegate address to the currently active paying
+    channel address for the given provider delegate.
+    """
+
+    paying_balances: PayingChannelBalances
+    """
+    Current off-chain balances within paying channels.
+    """
+
+    offers: Offers
+    """
+    Data encryption key offers.
+    """
+
+    idx_offer_by_key: IndexOfferByKey
+    """
+    Index of key offers by key ID (rather than offer ID, as the object table
+    is indexed by).
+    """
+
+    transaction: Transactions
+    """
+    """
+
+    @staticmethod
+    def attach(db):
+        """
+        Factory to create a schema from attaching to a database. The schema tables
+        will be automatically mapped as persistant maps and attached to the
+        database slots.
+
+        :param db: zlmdb.Database
+        :return: object of Schema
+        """
+        schema = Schema(db)
+
+        schema.consents = db.attach_table(Consents)
```

Comment:
sorry, after relooking at my issue description: this is a bug in my desc .. we need consents (read-only) from the market maker, but the on-chain submission is done in xbr network backend, and the on-chain information (consents) are replicated and written on a xbr worker at the worker level. hence `consents` (and `idx_consent_by_member_adr`) should be moved to schema `xbr` (from `xbrmm`).

the other tables are correctly placed in this schema, that is: channels, offers and transactions

Reply:
done that